src/Tmds.Ssh/GssapiWithMicCredential.cs

```diff
@@ -0,0 +1,282 @@
+// This file is part of Tmds.Ssh which is released under MIT.
+// See file LICENSE for full license details.
+
+using System;
+using System.Buffers;
+using System.Linq;
+using System.Net;
+using System.Net.Security;
+using System.Reflection;
+using System.Security.Principal;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Tmds.Ssh;
+
+public sealed class GssapiWithMicCredential : Credential
```

Comment:
What I'm not sure on is how this would look if `gssapi-keyex` is ever implemented. The `keyex` method introduces a new host verification process bundled with user authentication. Technically GSSAPI can also negotiate more than just Kerberos but I'm not sure if any server implementation really does anything other than Kerberos through it.

Reply:
The `Credential` derived types are meant to capture information for authentication and (in theory) they should be usable in different situation. The "worst case" requires introducing a new type of credential.

I'd like a descriptive user-friendly name. I don't think `GssapiWithMic` means much to users of this library. I assume `KerberosCredential` does.

Reply:
Makes sense, Kerberos is certainly more user friendly.

Reply:
Let's use `KerberosCredential`.